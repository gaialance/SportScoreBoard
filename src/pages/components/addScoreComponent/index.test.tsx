// import dependencies
import { render,screen , cleanup, fireEvent } from '@testing-library/react'

import '@testing-library/jest-dom/extend-expect'
import AddScoreComponent from '.'

afterEach(cleanup);

describe("rendering the NavBarComponent",()=>{
    it('making Snapsnot the landing page to compare',() => {
        const { container } = render(<AddScoreComponent leagueId={0} />)
        
        expect(container).toMatchSnapshot()
    })

    it("clicking on the create button and generating snapshot and cancel test",() => {
        const { container } = render(<AddScoreComponent leagueId={0} />)

        const btnCreate = screen.getByTestId("buttonTest")  

        expect(container).toMatchSnapshot()

        fireEvent.click(btnCreate);

        expect(container).toMatchSnapshot()

        const btnCancel = screen.getByTestId("cancelBtn")

        fireEvent.click(btnCancel);

        expect(container).toMatchSnapshot()
    })

    test("clicking on the create button and submit snapshot",() => {

        const { container } = render(<AddScoreComponent leagueId={0} />)

        expect(container).toMatchSnapshot()

        const btnCreate = screen.getByTestId("buttonTest")  

        fireEvent.click(btnCreate);

        expect(container).toMatchSnapshot()

        const btnSubmit = screen.getByTestId("submitBtn")

        fireEvent.click(btnSubmit);

        expect(container).toMatchSnapshot()
    })
})