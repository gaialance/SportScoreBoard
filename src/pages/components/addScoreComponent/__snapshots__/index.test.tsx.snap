// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering the NavBarComponent clicking on the create button and generating snapshot and cancel test 1`] = `
<div>
  <div
    class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1ib8r6l-MuiGrid-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium button css-1trkvrt-MuiButtonBase-root-MuiButton-root"
          data-testid="buttonTest"
          tabindex="0"
          type="button"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Create New
          </p>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`rendering the NavBarComponent clicking on the create button and generating snapshot and cancel test 2`] = `
<div
  aria-hidden="true"
>
  <div
    class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1ib8r6l-MuiGrid-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium button css-1trkvrt-MuiButtonBase-root-MuiButton-root"
          data-testid="buttonTest"
          tabindex="0"
          type="button"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Create New
          </p>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`rendering the NavBarComponent clicking on the create button and generating snapshot and cancel test 3`] = `
<div
  aria-hidden="true"
>
  <div
    class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1ib8r6l-MuiGrid-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium button css-1trkvrt-MuiButtonBase-root-MuiButton-root"
          data-testid="buttonTest"
          tabindex="0"
          type="button"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Create New
          </p>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`rendering the NavBarComponent clicking on the create button and submit snapshot 1`] = `
<div>
  <div
    class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1ib8r6l-MuiGrid-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium button css-1trkvrt-MuiButtonBase-root-MuiButton-root"
          data-testid="buttonTest"
          tabindex="0"
          type="button"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Create New
          </p>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`rendering the NavBarComponent clicking on the create button and submit snapshot 2`] = `
<div
  aria-hidden="true"
>
  <div
    class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1ib8r6l-MuiGrid-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium button css-1trkvrt-MuiButtonBase-root-MuiButton-root"
          data-testid="buttonTest"
          tabindex="0"
          type="button"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Create New
          </p>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`rendering the NavBarComponent clicking on the create button and submit snapshot 3`] = `
<div
  aria-hidden="true"
>
  <div
    class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1ib8r6l-MuiGrid-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium button css-1trkvrt-MuiButtonBase-root-MuiButton-root"
          data-testid="buttonTest"
          tabindex="0"
          type="button"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Create New
          </p>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`rendering the NavBarComponent making Snapsnot the landing page to compare 1`] = `
<div>
  <div
    class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1ib8r6l-MuiGrid-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium button css-1trkvrt-MuiButtonBase-root-MuiButton-root"
          data-testid="buttonTest"
          tabindex="0"
          type="button"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Create New
          </p>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
